bokeh/models/plots.py

```diff
@@ -371,17 +385,50 @@ def _check_snapped_toolbar_and_axis(self):
     The (default) data range of the vertical dimension of the plot.
     """)
 
-    x_mapper_type = Either(Auto, String, help="""
-    What kind of mapper to use to convert x-coordinates in data space
+    @classmethod
+    def _scale(cls, scale):
+        if scale == "auto":
+            return Auto
+        if scale == "categorical":
+            return CategoricalScale()
+        elif scale == "linear":
+            return LinearScale()
+        elif scale == "log":
+            return LogScale()
+        else:
+            raise ValueError("Unknown mapper_type")
+
+    @property
+    def x_mapper_type(self):
+        deprecated((0, 12, 6), "x_mapper_type", "x_scale")
+        return self.x_scale
+
+    @x_mapper_type.setter
+    def x_mapper_type(self, mapper_type):
+        deprecated((0, 12, 6), "x_mapper_type", "x_scale")
+        self.x_scale = self._scale(mapper_type)
+
+    @property
+    def y_mapper_type(self):
+        deprecated((0, 12, 6), "y_mapper_type", "y_scale")
+        return self.y_scale
+
+    @y_mapper_type.setter
+    def y_mapper_type(self, mapper_type):
+        deprecated((0, 12, 6), "y_mapper_type", "y_scale")
+        self.y_scale = self._scale(mapper_type)
+
+    x_scale = Either(Auto, Instance(Scale), help="""
```

Comment:
Yah it kind of sucks but we definitely need a "non scale thing" to indicate "choose a scale for me" OTOH what about just `None` for that purpose? It's already accepted... I'm not opposed to `Auto`, either way the "auto" behavior should be be the default since it is the current default.